src/Wrapper.js

```diff
@@ -311,6 +333,102 @@ class Wrapper extends React.PureComponent {
   }
 }
 
+const SwipeContainer = (() => {
+  let _progress = 0
+
+  const SwipeContainer = ({
+    children,
+    enabled,
+    menuPanelOpened,
+    onMenuPanelClose,
+    onMenuPanelOpen,
+    width,
+  }) => {
+    const oneThirdWindowWidth = width / 3
+
+    if (!enabled) {
+      return <Container>{children(Number(menuPanelOpened))}</Container>
+    }
+
+    return (
+      <Gesture passive={{ passive: true }}>
+        {({
+          delta: [xDelta, yDelta],
+          direction: [xDir, yDir],
+          down,
+          event,
+          initial: [xInitial],
+          xy: [x],
+        }) => {
+          if (!down && _progress > 0 && _progress < 1) {
+            _progress = !menuPanelOpened
```

Comment:
Nice.
Most of this code came from when I was testing with a different threshold and then I kept staring at it and could not figure a way to optimize it, especially that first `if` with the `!menuPanelOpened`. I do prefer to not add the `Math.round` and keep the `THRESHOLD` to make this breakpoint more explicit.
The `setTimeout` is needed to update the `Wrapper` state; it holds a state because there is another way to `open/close` the `MenuPanel` via clicks on buttons.